core/arch/arm/mm/mobj.c

```diff
@@ -639,6 +663,68 @@ TEE_Result mobj_reg_shm_unmap(struct mobj *mobj)
 	return TEE_SUCCESS;
 }
 
+TEE_Result mobj_reg_shm_inc_map(struct mobj *mobj)
+{
+	TEE_Result res = TEE_SUCCESS;
+	struct mobj_reg_shm *r = to_mobj_reg_shm_may_fail(mobj);
+
+	if (!r)
+		return TEE_ERROR_GENERIC;
+
+	if (refcount_inc(&r->mapcount))
+		return TEE_SUCCESS;
```

Comment:
shouldn't the refcount be under the same non preempible context as the alloc/map sequence below (inside `spin_lock_xsave()/xrestore()` bounds)?

Same question for function  `mobj_reg_shm_dec_map()` below.

Reply:
Not needed. `refcount_inc()` and `refcount_dec()` are atomic and as long as the counter doesn't reach 0 no further update is needed. An update requires a lock though, and then we check if another thread has beat us to do the update.